### PR TITLE
feat: update oskp status screen

### LINKED ISSUE #753: time is missing on this screen
cf. figma

IMG_8351

### PR DESCRIPTION
**Pull request by justinenerio**
## Changes

- added timestamp when links are generated and resolved (sucess/cancelled)
- updated items (figma)

## Related issues

Fixes #753

## Videos

<details><summary>Normal Flow</summary>

https://user-images.githubusercontent.com/28533130/218663592-23ce3fb7-af07-4bea-9cac-094001fbc42c.mp4

https://user-images.githubusercontent.com/28533130/218792796-0fa37197-9a74-4792-b1ac-f4db8e6db231.mp4

</details>

<details><summary>Cancel</summary>

https://user-images.githubusercontent.com/28533130/218663682-a3cb4ee5-b242-4220-a36f-603d8a1fc7a6.mp4

</details>

<details><summary>Activity History</summary>

https://user-images.githubusercontent.com/28533130/218663755-e021a821-c36b-4995-b695-d832fb859e78.mp4

</details>

## Checklist

- [x] PR is ready for review (if not, it should be a draft).
- [x] PR title follows [Conventional Commits][1] guidelines.
- [x] Screenshots/video added.
- [ ] Tests added.
- [x] Self-review done.

[1]: https://www.conventionalcommits.org/en/v1.0.0/

### CODE REVIEW

File: packages/espressocash_app/lib/features/outgoing_split_key_payments/models/outgoing_split_key_payment.dart [hunk 1]
```diff
@@ -14,6 +14,8 @@ class OutgoingSplitKeyPayment with _$OutgoingSplitKeyPayment {
     required CryptoAmount amount,
     required DateTime created,
     required OSKPStatus status,
+    DateTime? linksGeneratedAt,
+    DateTime? resolvedAt,
```
Comment (ookami-kb) [inline packages/espressocash_app/lib/features/outgoing_split_key_payments/models/outgoing_split_key_payment.dart:18]: You don't need to store `resolvedAt` here, it will be on the `canceled`/`withdrawn` statuses, and they are terminal.

File: packages/espressocash_app/lib/features/outgoing_split_key_payments/models/outgoing_split_key_payment.dart [hunk 2]
```diff
@@ -47,17 +49,20 @@ class OSKPStatus with _$OSKPStatus {
     required Uri link2,
     Uri? qrLink,
     required EscrowPrivateKey escrow,
+    DateTime? timestamp,
```
Comment (ookami-kb) [inline packages/espressocash_app/lib/features/outgoing_split_key_payments/models/outgoing_split_key_payment.dart:52]: Here, vice versa, you don't need a timestamp on the status, you already have a field on the `OutgoingSplitKeyPayment` model.

File: packages/espressocash_app/lib/features/outgoing_split_key_payments/models/outgoing_split_key_payment.dart [hunk 3]
```diff
@@ -47,17 +49,20 @@ class OSKPStatus with _$OSKPStatus {
...
   }) = OSKPStatusLinksReady;
 
   /// Money are withdrawn from the escrow by someone, but not by the sender. The
   /// payment is complete.
   const factory OSKPStatus.withdrawn({
     required String txId,
+    DateTime? timestamp,
```
Comment (ookami-kb) [inline packages/espressocash_app/lib/features/outgoing_split_key_payments/models/outgoing_split_key_payment.dart:59]: Should not be nullable here.

File: packages/espressocash_app/lib/features/outgoing_split_key_payments/models/outgoing_split_key_payment.dart [hunk 4]
```diff
@@ -47,17 +49,20 @@ class OSKPStatus with _$OSKPStatus {
...
   }) = OSKPStatusWithdrawn;
 
   /// Money are withdrawn by the sender. The payment is complete.
   const factory OSKPStatus.canceled({
     required String? txId,
+    DateTime? timestamp,
```
Comment (ookami-kb) [inline packages/espressocash_app/lib/features/outgoing_split_key_payments/models/outgoing_split_key_payment.dart:65]: Should not be nullable as well.

File: packages/espressocash_app/lib/features/outgoing_split_key_payments/src/bl/tx_ready_watcher.dart [hunk 5]
```diff
@@ -91,7 +97,7 @@ class TxReadyWatcher {
           .flatMap(streamSignatures)
           .where((event) => event.length == 2)
           .map((details) => details.first)
-          .map((tx) => tx.transaction as ParsedTransaction),
+          .map((tx) => tx),
```
Comment (ookami-kb) [inline packages/espressocash_app/lib/features/outgoing_split_key_payments/src/bl/tx_ready_watcher.dart:100]: Useless line.

**Review by ookami-kb — CHANGES_REQUESTED** [review]

**Comment by ookami-kb** [comment]
@justinenerio please resolve conflicts.

**Comment by justinenerio** [comment]
@ookami-kb conflicts are resolved

File: packages/espressocash_app/lib/features/outgoing_split_key_payments/src/bl/repository.dart [hunk 6]
```diff
@@ -178,6 +181,7 @@ extension on OSKPStatusDto {
     final link3 = row.link3?.let(Uri.tryParse);
     final cancelTx = row.cancelTx?.let(SignedTx.decode);
     final cancelTxId = row.cancelTxId;
+    final resolvedAt = row.resolvedAt ?? DateTime.now();
```
Comment (ookami-kb) [inline packages/espressocash_app/lib/features/outgoing_split_key_payments/src/bl/repository.dart:184]: Don't use `DateTime.now()` as a fallback. Otherwise, it will be different on every loading.

It's better to make the field nullable in that case (but still required), and in the future, when all the clients are migrated, we can move it back to non-nullable.

**Review by ookami-kb — CHANGES_REQUESTED** [review]

File: packages/espressocash_app/moor_schemas/moor_schema_v29.json [hunk 7]
```diff
@@ -1 +1 @@
...getter_name":"id","moor_type":"string","nullable":false,"customConstraints":null,"default_dart":null,"default_client_dart":null,"dsl_features":[]},{"name":"created","getter_name":"created","moor_type":"dateTime","nullable":false,"customConstraints":null,"default_dart":null,"default_client_dart":null,"dsl_features":[]},{"name":"private_key","getter_name":"privateKey","moor_type":"string","nullable":false,"customConstraints":null,"default_dart":null,"default_client_dart":null,"dsl_features":[]},{"name":"status","getter_name":"status","moor_type":"int","nullable":false,"customConstraints":null,"default_dart":null,"default_client_dart":null,"dsl_features":[],"type_converter":{"dart_expr":"const EnumIndexConverter<ISLPStatusDto>(ISLPStatusDto.values)","dart_type_name":"ISLPStatusDto"}},{"name":"tx","getter_name":"tx","moor_type":"string","nullable":true,"customConstraints":null,"default_dart":null,"default_client_dart":null,"dsl_features":[]},{"name":"tx_id","getter_name":"txId","moor_type":"string","nullable":true,"customConstraints":null,"default_dart":null,"default_client_dart":null,"dsl_features":[]}],"is_virtual":false,"without_rowid":false,"constraints":[],"explicit_pk":["id"]}}]}
```
Comment (ookami-kb) [inline packages/espressocash_app/moor_schemas/moor_schema_v29.json:1]: This file shouldn't be included in the PR.

**Review by ookami-kb — CHANGES_REQUESTED** [review]

**Review by ookami-kb — APPROVED** [review]
